### DISCUSSION TITLE
How to debug adonis typescript source code in adonis v5 application?

### DISCUSSION THREAD

**Discussion by zhaozhao** [adonisjs/core#1431] [Help] [answered]
I aim to learn how adonis works by debugging an adonis v5 application. I'm more interested in how adonis, the framework itself, works, rather than using the framework to build web applications. 

The stack got from debugging my tiny hello-world adonis application contains only the generated javascript source code, not the origin typescript source code. 
![image](https://user-images.githubusercontent.com/3445700/89749061-cdefa280-daf8-11ea-81ea-8eb3209afa6e.png)

It seems that the adonis npm package does not contain the typescript source code or the source map.

It would be quiet handy if I can get the code stack contains the original ts source code rather than the generated js code.

Is this possible? What should I do to accomplish that?

**Comment by thetutlage** [comment, marked as answer]
I believe, it is a standard practice to not ship the typescript code in the packages and just ship the compiled Javascript code, otherwise you will be doubling the package size.

Is there any reason that you prefer looking at the Typescript code and not Javascript, since the compiled JS is equally readable

**Reply by zhaozhao** [reply]
No specific reason. Just it would be easier to reason about how does adonis work by debugging typescript source code which has better code navigation support and type info. Currently, I can only get the type declaration from the symbols imported from the adonis framework. I have to try to guess which file seems to be the underlying implementation javascript file, which is not quite friendly to figure out how the ORM works, how the injection dependency works.

Thx for the reply.
